controller_manager:
  ros__parameters:
    update_rate: 100  # Hz
    
    # ros2_controllers
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster


joint_trajectory_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6
      - joint_7

    command_interfaces:
      - position
      # - velocity

    state_interfaces:
      - position
      - velocity
      # - acceleration
      # - effort

    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true